lnd.go

```diff
@@ -208,19 +208,35 @@ func lndMain() error {
 		defer macaroonService.Close()
 	}
 
+	var (
+		privateWalletPw = []byte("hello")
+		publicWalletPw  = []byte("public")
+		birthday        time.Time
+		recoveryWindow  uint32
+	)
+
 	// We wait until the user provides a password over RPC. In case lnd is
 	// started with the --noencryptwallet flag, we use the default password
 	// "hello" for wallet encryption.
-	privateWalletPw := []byte("hello")
-	publicWalletPw := []byte("public")
 	if !cfg.NoEncryptWallet {
-		privateWalletPw, publicWalletPw, err = waitForWalletPassword(
-			cfg.RPCListeners, cfg.RESTListeners, serverOpts, proxyOpts,
-			tlsConf, macaroonService,
+		walletInitParams, err := waitForWalletPassword(
```

Comment:
what will happen if `--noencryptwallet` is specified? Atm I don't think we even give users a seed (hence they won't be able to recover the funds), but this is not very intuitive. We should probably let a user run with `--noencryptwallet` and recover from a seed, or deprecate `--noencryptwallet` (at least on mainnet).

Reply:
yeah we don't have rescans for unencrypted wallets, because the users don't have seeds.
currently the recovery window is initialized as 0, and can only be set when `--noencryptwallet` is not provided, so it will just sync as normal